Mathlib/CategoryTheory/MorphismProperty/Comma.lean

```diff
@@ -203,6 +211,102 @@ def forgetFullyFaithful : (forget L R P ⊤ ⊤).FullyFaithful where
 instance : (forget L R P ⊤ ⊤).Full :=
   Functor.FullyFaithful.full (forgetFullyFaithful L R P)
 
+section
+
+variable {L R}
+
+@[simp]
+lemma eqToHom_left {X Y : P.Comma L R Q W} (h : X = Y) :
+    (eqToHom h).left = eqToHom (by rw [h]) := by
+  subst h
+  rfl
+
+@[simp]
+lemma eqToHom_right {X Y : P.Comma L R Q W} (h : X = Y) :
+    (eqToHom h).right = eqToHom (by rw [h]) := by
+  subst h
+  rfl
+
+end
+
+section Functoriality
+
+variable {L R P Q W}
+variable {L₁ L₂ L₃ : A ⥤ T} {R₁ R₂ R₃ : B ⥤ T}
+
+/-- Lift a functor `F : C ⥤ Comma L R` to the subcategory `P.Comma L R Q W` under
+suitable assumptions on `F`. -/
+def lift {C : Type*} [Category C] (F : C ⥤ Comma L R)
+    (hP : ∀ X, P (F.obj X).hom)
+    (hQ : ∀ {X Y} (f : X ⟶ Y), Q (F.map f).left)
+    (hW : ∀ {X Y} (f : X ⟶ Y), W (F.map f).right) :
+    C ⥤ P.Comma L R Q W where
+  obj X :=
+    { __ := F.obj X
+      prop := hP X }
+  map {X Y} f :=
+    { __ := F.map f
+      prop_hom_left := hQ f
+      prop_hom_right := hW f }
+
+variable (R) in
+/-- A natural transformation `L₁ ⟶ L₂` induces a functor `P.Comma L₂ R Q W ⥤ P.Comma L₁ R Q W`. -/
+@[simps obj_left obj_right]
+def mapLeft (l : L₁ ⟶ L₂) (hl : ∀ X : P.Comma L₂ R Q W, P (l.app X.left ≫ X.hom)) :
+    P.Comma L₂ R Q W ⥤ P.Comma L₁ R Q W where
+  obj X :=
+    { left := X.left
+      right := X.right
+      hom := l.app X.left ≫ X.hom
+      prop := hl X }
+  map {X Y} f :=
+    { left := f.left
+      right := f.right
+      prop_hom_left := f.prop_hom_left
+      prop_hom_right := f.prop_hom_right }
```

Comment:
The following definitions is defeq to yours and it also generates the expected simp lemmas:
```suggestion
@[simps! obj_left obj_right map_left map_right]
def mapLeft(l : L₁ ⟶ L₂) (hl : ∀ X : P.Comma L₂ R Q W, P (l.app X.left ≫ X.hom)) :
    P.Comma L₂ R Q W ⥤ P.Comma L₁ R Q W :=
  lift (forget _ _ _ _ _ ⋙ CategoryTheory.Comma.mapLeft R l) hl
    (fun f ↦ f.prop_hom_left) (fun f ↦ f.prop_hom_right)
```
The same should be done for `mapRight`.